### ISSUE TITLE
Bug when updating to version 2.0

### ISSUE THREAD

**Issue by Dominiquini** [kryojenik/LTN_Combinator_Modernized#28]
All my LTN Combinators formerly configured as providers are now configured as providers and requesters (both checkbox enabled!).
I could select all my provider stations on the map and switching one by one, but I hope that can be fix with a new version!

Thanks.

**Comment by kryojenik** [comment]
Yeah, I kind of anticipated this one and probaly should have created a note in the changelog.

This is a cosmetic thing. LTN doesn't have a control to make stations just provider or just requester. They are alway both. You can effectively disable the ability of a station to provide or request things by assigning a really high threshold, and that is what is done when you disable the Requester or Provider service in 2.0 version of the combinator.

The previous version only inferred Requester or Provider by the presence of negative item signal or a high provider threshold threshold. If you did not have a negative item / fluid signal or there was a high provider signal the combinator would display as a provider. In reality, simply adding a negative signal would have resulted in both provider and requester being checked the next time you opened it. This led to lots of confusion (as well as feature requests) all stemming from the fact that LTN does not have distinct provider and requester stations.

The changes I made in 2.0 will take a little adjustment in thinking, but provide a more accurate representation of how the station is configured (actuall control signals) and some additional utility by being able to enable and disable the "service" without loosing threshold settings.

That said, I will add a command to uncheck "Requester" on combinators without negative signals.

**Comment by Dominiquini** [comment]
Seen the migration code, maybe the bug is because it looks for requester_thresholds signals >= 50000000 to disable the requester checkbox from a provider station! But my providers don't emit any requester_threshold signal when I was using the version "1.0.14"!

**Comment by Dominiquini** [comment]
> Yeah, I kind of anticipated this one and probaly should have created a note in the changelog.
> 
> This is a cosmetic thing. LTN doesn't have a control to make stations just provider or just requester. They are alway both. You can effectively disable the ability of a station to provide or request things by assigning a really high threshold, and that is what is done when you disable the Requester or Provider service in 2.0 version of the combinator.
> 
> The previous version only inferred Requester or Provider by the presence of negative item signal or a high provider threshold threshold. If you did not have a negative item / fluid signal or there was a high provider signal the combinator would display as a provider. In reality, simply adding a negative signal would have resulted in both provider and requester being checked the next time you opened it. This led to lots of confusion (as well as feature requests) all stemming from the fact that LTN does not have distinct provider and requester stations.
> 
> The changes I made in 2.0 will take a little adjustment in thinking, but provide a more accurate representation of how the station is configured (actuall control signals) and some additional utility by being able to enable and disable the "service" without loosing threshold settings.
> 
> That said, I will add a command to uncheck "Requester" on combinators without negative signals.

I wrote the previous message, but forget to "send"!
Maybe it's possible to change the migration code and disable the "requester" checkbox if the combinator don't have any requester_threshold configured?

Thanks.
